django/db/models/fields/__init__.py

```diff
@@ -270,16 +272,32 @@ def is_value(value, accept_promise=True):
                     for value, human_name in group_choices
                 ):
                     break
+                if self.max_length is not None:
+                    max_length = max(
+                        max_length,
+                        *(len(value) for value, human_name in group_choices if isinstance(value, str)),
+                    )
             except (TypeError, ValueError):
                 # No groups, choices in the form [value, display]
                 value, human_name = group_name, group_choices
                 if not is_value(value) or not is_value(human_name):
                     break
+                if self.max_length is not None and isinstance(value, str):
+                    max_length = max(max_length, len(value))
```

Comment:
Why we use `max()` here? :thinking: i.e.
```python
max_length = len(value)
```
works fine for me.

Reply:
Don't forget that there is a loop `for choices_group in self.choices:` and we need the longest value, not the length of the last value.